app/controllers/spree/admin/base_controller.rb

```diff
@@ -19,15 +19,18 @@ class BaseController < ApplicationController
 
       before_action :authorize_admin
       before_action :set_locale
-      before_action :warn_invalid_order_cycles, if: :html_request?
+      before_action :warn_invalid_order_cycles, if: :page_load_request?
 
       # Warn the user when they have an active order cycle with hubs that are not ready
       # for checkout (ie. does not have valid shipping and payment methods).
       def warn_invalid_order_cycles
-        return if flash[:notice].present?
+        return if session[:displayed_order_cycle_warning]
 
         warning = OrderCycles::WarningService.new(spree_current_user).call
-        flash[:notice] = warning if warning.present?
+        return if warning.blank?
+
+        flash.now[:notice] = warning
+        session[:displayed_order_cycle_warning] = true
```

Comment:
Does this mean that we display the message only once?

Reply:
Yes, only once during the user's login session. If he re-logins then the warning will again pop up.

Reply:
Ref: https://openfoodnetwork.slack.com/archives/C01CXQNJ1J6/p1723641747029789?thread_ts=1723641747.029789&cid=C01CXQNJ1J6